modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildSelectSuitesModal.tsx

```diff
@@ -12,58 +12,130 @@
  * details.
  */
 
-import React, {useState} from 'react';
+import React, {Dispatch, SetStateAction, useState} from 'react';
 import {useParams} from 'react-router-dom';
 
 import Form from '../../../../components/Form';
-import Container from '../../../../components/Layout/Container';
 import ListView from '../../../../components/ListView';
 import Modal from '../../../../components/Modal';
 import {withVisibleContent} from '../../../../hoc/withVisibleContent';
 import {FormModalOptions} from '../../../../hooks/useFormModal';
 import i18n from '../../../../i18n';
 import {filters} from '../../../../schema/filter';
+import fetcher from '../../../../services/fetcher';
+import {APIResponse, TestraySuiteCase} from '../../../../services/rest';
+import {getUniqueList} from '../../../../util';
 import {searchUtil} from '../../../../util/search';
+import SelectCase from '../../Suites/modal/SelectCase';
+
+type ModalType = {
+	type: 'select-cases' | 'select-suites';
+};
 
 type BuildSelectSuitesModalProps = {
+	displayTitle?: boolean;
 	modal: FormModalOptions;
+	setModalType: Dispatch<SetStateAction<ModalType>>;
+	type: 'select-cases' | 'select-suites';
 };
 
 const BuildSelectSuitesModal: React.FC<BuildSelectSuitesModalProps> = ({
-	modal: {observer, onClose, onSave, visible},
+	displayTitle = false,
+	modal: {modalState, observer, onClose, onSave, visible},
+	setModalType,
+	type,
 }) => {
-	const [state, setState] = useState<any>({});
+	const [state, setState] = useState<any>([]);
```

Comment:
Find a better name and types

```suggestion
	const [state, setState] = useState<any>([]);
```